vault_program/src/mint_to.rs

```diff
@@ -69,6 +71,16 @@ pub fn process_mint(
     vault.check_vrt_mint(vrt_mint.key)?;
     vault.check_update_state_ok(Clock::get()?.slot, config.epoch_length())?;
 
+    if depositor.key.eq(vault_info.key) {
+        msg!("Depositor cannot be the vault");
+        return Err(VaultError::InvalidDepositor.into());
+    }
+
+    if depositor_token_account.key.eq(vault_token_account.key) {
+        msg!("Depositor token account cannot be the vault token account");
+        return Err(VaultError::InvalidDepositTokenAccount.into());
+    }
+
     let MintSummary {
         vrt_to_depositor,
         vrt_to_fee_wallet,
```

Comment:
can you also add checks after the invokes to ensure the vault mint amount = the token mint supply + other things mentioned in the audit report?

Reply:
10-4